### PR TITLE
Fixing and adding support for custom linkAttribute classes

### PR DESCRIPTION
**Pull request by arfaram**
| Question      | Answer
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Doc needed?   | no

Jira Ticket: [EZP-28928](https://jira.ez.no/browse/EZP-28928)

#### Checklist:
- [x] Ready for Code Review

### CODE REVIEW

**Comment by lserwatka** [comment]
@arfaram could you re-base this with 1.0 branch?

File: src/bundle/Resources/views/parts/menu/top_menu.html.twig [hunk 1]
```diff
@@ -22,7 +22,7 @@
     {% set linkAttributes = item.linkAttributes|merge({
         'href': item.uri is not empty ? item.uri : '#' ~ item.name,
         'aria-controls': item.name,
-        'class': item.linkAttributes|default('') ~ (' nav-link' ~ (matcher.isAncestor(item, options.matchingDepth) ? ' ' ~ options.currentClass))|trim,
+        'class': ('nav-link' ~ (matcher.isAncestor(item, options.matchingDepth) ? ' ' ~ options.currentClass|trim)) ~  (item.linkAttributes.class is defined ?  ' ' ~ item.linkAttributes.class|trim),
```
Comment (webhdx) [inline src/bundle/Resources/views/parts/menu/top_menu.html.twig]: @arfaram Isn't `item.linkAttributes.class|default('')` => `item.linkAttributes.class|default('')` a sufficient fix?

File: src/bundle/Resources/views/parts/menu/top_menu_2nd_level.html.twig [hunk 2]
```diff
@@ -27,7 +27,7 @@
     {% set linkAttributes = item.linkAttributes|merge({
         'href': item.uri,
         'aria-controls': item.name,
-        'class': 'nav-link' ~ (matcher.isCurrent(item) ? ' ' ~ options.currentClass)
+        'class': ('nav-link' ~ (matcher.isCurrent(item) ? ' ' ~ options.currentClass|trim)) ~ (item.linkAttributes.class is defined ?  ' ' ~ item.linkAttributes.class|trim)
```
Comment (webhdx) [inline src/bundle/Resources/views/parts/menu/top_menu_2nd_level.html.twig]: As above:
`'class': ('nav-link' ~ (matcher.isCurrent(item) ? ' ' ~ options.currentClass|trim)) ~ ' ' ~ item.linkAttributes.class|default(''))|trim`

Haven't tested it though.

File: src/bundle/Resources/views/parts/menu/top_menu.html.twig (hunk 1 above)
Comment (arfaram) [reply src/bundle/Resources/views/parts/menu/top_menu.html.twig]: @webhdx also works but I need to add a blank before when it is defined.

**Comment by lserwatka** [comment]
@webhdx @arfaram what is the status here? Could we move forward with this change, either merge it or close this PR.

**Comment by andrerom** [comment]
I guess we lack a suggestion on how to better do this and make sure there is space between the classes. array & implode ? :)

**Comment by arfaram** [comment]
@webhdx  @lserwatka I've added the Jira Ticket is the description

**Comment by lserwatka** [comment]
@arfaram thank you, but what is the status of this code, is this ready?

**Comment by arfaram** [comment]
The last comment of @andrerom can I not identify it. The PR still works with latest 2.0.2. @webhdx you could have the last word before merging

**Pull request merged**